src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -128,17 +138,9 @@ public AddressBook load() throws StorageOperationException {
             }
             return loaded.toModelType();
 
-        /* Note: Here, we are using an exception to create the file if it is missing. However, we should minimize
-         * using exceptions to facilitate normal paths of execution. If we consider the missing file as a 'normal'
-         * situation (i.e. not truly exceptional) we should not use an exception to handle it.
-         */
-
-        // create empty file if not found
         } catch (FileNotFoundException fnfe) {
-            final AddressBook empty = new AddressBook();
-            save(empty);
-            return empty;
-
+            assert false : "not possible: is already handled earlier";
```

Comment:
Since there is a specific exception for file not found, why not handle the logic here (that is reutrn the empty addressbook here) instead of adding an extra check above?